app/V1Module/presenters/AssignmentSolutionsPresenter.php

```diff
@@ -189,17 +192,28 @@ public function checkSetBonusPoints(string $id) {
    * Set new amount of bonus points for a solution
    * @POST
    * @Param(type="post", name="bonusPoints", validation="numericint", description="New amount of bonus points, can be negative number")
-   * @Param(type="post", name="overriddenPoints", validation="null|numericint", description="Overrides points assigned to solution by the system")
+   * @Param(type="post", name="overriddenPoints", description="Overrides points assigned to solution by the system")
    * @param string $id Identifier of the submission
    * @throws NotFoundException
+   * @throws InvalidArgumentException
    */
   public function actionSetBonusPoints(string $id) {
     $newBonusPoints = $this->getRequest()->getPost("bonusPoints");
     $overriddenPoints = $this->getRequest()->getPost("overriddenPoints");
     $solution = $this->assignmentSolutions->findOrThrow($id);
 
     $solution->setBonusPoints($newBonusPoints);
-    $solution->setOverriddenPoints($overriddenPoints === "" ? null : $overriddenPoints);
+
+    // TODO: validations 'null|numericint' for overridenPoints cannot be used, because null is converted to empty string,
+    // TODO: which immediately breaks stated validation... in the future, this behaviour has to change
+    // TODO: lucky third TODO
```

Comment:
Actually, you have four TODOs there, which breaks the luck...

Reply:
Darn it